src/Appl/Ranger.f90

```diff
@@ -79,7 +82,118 @@ subroutine singlerange(partcls,nplc,nptot,range,center)
         enddo
       endif
 
-      end subroutine singlerange
+      end subroutine singlerange_standard
+
+    !--------------------------------------------------------------------------------------
+    !> @brief
+    !> find the range (minimum and maxium of a single bunch) and the centroid
+    !> of a single beam bunch, filtering for a certain momentum range.
+    !--------------------------------------------------------------------------------------
+    subroutine singlerange_filtered(partcls,nplc,nptot,range,center,&
+                                    fmin, fmax, frange, fcenter, fnptot)
```

Comment:
The new subroutine contains all the same code as the original subroutine, but with additional ranges and particle counts calculated for the particles filtered by energy range.

The results for `singlerange_filtered(partcls,nplc,nptot,range,center,…)` should match exactly the results for `singlerange_standard(partcls,nplc,nptot,range,center)`, but also return the additional values `fmin, fmax, frange, fcenter, fnptot`.